### DISCUSSION TITLE
Sorry for the dumb question

### DISCUSSION THREAD

**Discussion by kparadine** [NuSkooler/enigma-bbs#363] [Q&A] [answered]
I have had the BBS running nicely on two different systems - one local, one an Ubuntu VM in Azure.

My question is how to write a wrapper shell script so I can use a crontab to spawn the BBS.  I've run MUDs for many years amongst other things in such environments.  A script like so:

#!/bin/sh
#
# Enigma BBS startup script
# node ./main.js is what it looks like in ps

TEST=`ps x -j | awk '/node .\/main.js/ && !/awk/ {print $1}'`
echo Enigma pid: "$TEST"
if test -z $TEST; then
        cd /home/AzureUser/enigma-bbs
        ./main.js
        echo "Starting Enigma.";
        else
        echo "Not starting Enigma.";
fi

...does not work.  It'll work to spawn the bbs from an interactive session, and prevent it from respawning from another console window, but my whole point is to spawn Enigma non-interactively.  I put it in the crontab because that will respawn it if it crashes, which is a problem i've had with other daemons over time.

I'm probably just being dumb about how node works.

Thanks for your attention.

**Comment by kparadine** [comment, marked as answer]
And I just saw PM2 so i'm investigating that now.  Maybe solves my issue.

**Reply by NuSkooler** [reply]
@kparadine were you able to get this resolved?

**Reply by kparadine** [reply]
Absolutely.  PM2 worked like a charm, thanks.  It's the right way to do what I was trying to do with node.
